nemo/collections/asr/modules/conv_asr.py

```diff
@@ -399,9 +411,14 @@ def forward(self, encoder_output):
         embs = []
 
         for layer in self.emb_layers:
-            pool, emb = layer(pool), layer[:2](pool)
+            pool, emb = layer(pool), layer[: self.emb_id](pool)
             embs.append(emb)
 
+        if self.angular:
+            for W in self.final.parameters():
+                _ = F.normalize(W, p=2, dim=1)
```

Comment:
https://pytorch.org/docs/master/nn.functional.html#torch.nn.functional.normalize

F.normalize is not an inplace op unless you use `out=`, so whats the point of this loop then?

Reply:
It just normalizes the weights before calculating loss. I missed `W =` here, thanks Som